siad/update.go

```diff
@@ -31,50 +31,50 @@ func newerVersion(version string) bool {
 		// inputs are trusted, so no need to check the error
 		ni, _ := strconv.Atoi(nums[i])
 		Ni, _ := strconv.Atoi(NUMS[i])
-		if ni > Ni {
-			return true
+		if ni != Ni {
+			return ni > Ni
 		}
```

Comment:
ni?

Please try to avoid using two letter names. I have no idea what this variable means.